app/services/sign_in/access_token_audience_generator.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module SignIn
+  class AccessTokenAudienceGenerator
+    SHARED_SESSION_CLIENT_IDS_CACHE_KEY = 'sis_shared_sessions_client_ids'
+
+    def initialize(client_id:)
+      @client_id = client_id
```

Comment:
What if you passed the `ClientConfig` itself as a param? Then you could do `return shared_session_client_ids if client_config.shared_session` in the `generate_audience` function and might be a slightly more straightforward way to show that logic

Reply:
yeah good idea